### ISSUE TITLE
🐛 [firebase_auth] Apple Sign In uses email & full name scope even when not set

### ISSUE THREAD

**Issue by nilsreichardt** [firebase/flutterfire#9471]
## Bug report

Even when using an empty list of scopes, email and full name is still requested.

**Describe the bug**

https://user-images.githubusercontent.com/24459435/188511812-b1db762b-852a-49bc-9df0-0e205f260bed.mov

As you can see in the video, my full name and my email is still requested.

### Steps to reproduce

1. Clone this repository
2. Go to `flutterfire/packages/firebase_auth/firebase_auth/example/lib/main.dart`
4. Run the example of `firebase_auth`
5. Sign in with Apple

```dart
Future<void> _signInWithApple() async {
    final appleProvider = AppleAuthProvider(); // No scopes are added.

    if (kIsWeb) {
      // Once signed in, return the UserCredential
      await _auth.signInWithPopup(appleProvider);
    } else {
      await _auth.signInWithAuthProvider(appleProvider);
    }
  }
```

### Expected behavior

When adding no scopes, I would expect that Apple Sign In is executed with no scopes (full name and email should not be requested).

### Sample project

https://github.com/firebase/flutterfire/tree/master/packages/firebase_auth/firebase_auth/example

---

## Additional context

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
Flutter 3.0.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f1875d570e (8 weeks ago) • 2022-07-13 11:24:16 -0700
Engine • revision e85ea0e79c
Tools • Dart 2.17.6 • DevTools 2.12.2
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.17.6
Flutter SDK 3.0.5
firebase_auth_example 0.0.0

dependencies:
- firebase_auth 3.7.0 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_core 1.21.1 [firebase_core_platform_interface firebase_core_web flutter meta]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_signin_button 2.0.0 [flutter font_awesome_flutter]
- github_sign_in 0.0.5-dev.4 [flutter http webview_flutter url_launcher]
- google_sign_in 5.3.1 [flutter google_sign_in_android google_sign_in_ios google_sign_in_platform_interface google_sign_in_web]
- twitter_login 4.2.3 [flutter crypto http]

dev dependencies:
- drive 1.0.0-1.0.nullsafety.1 [test_api flutter_test flutter_driver stack_trace flutter]
- flutter_driver 0.0.0 [file flutter flutter_test fuchsia_remote_debug_protocol path meta vm_service webdriver archive async boolean_selector characters charcode clock collection crypto matcher material_color_utilities platform process source_span stack_trace stream_channel string_scanner sync_http term_glyph test_api typed_data vector_math]
- http 0.13.4 [async http_parser meta path]
- test 1.21.1 [analyzer async boolean_selector collection coverage http_multi_server io js node_preamble package_config path pool shelf shelf_packages_handler shelf_static shelf_web_socket source_span stack_trace stream_channel typed_data web_socket_channel webkit_inspection_protocol yaml test_api test_core]

dependency overrides:
- firebase_auth 3.7.0 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_auth_platform_interface 6.6.0 [collection firebase_core flutter meta plugin_platform_interface]
- firebase_auth_web 4.3.0 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core 1.21.1 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_core_platform_interface 4.5.1 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 1.7.2 [firebase_core_platform_interface flutter flutter_web_plugins js meta]

transitive dependencies:
- _fe_analyzer_shared 40.0.0 [meta]
- analyzer 4.1.0 [_fe_analyzer_shared collection convert crypto glob meta package_config path pub_semver source_span watcher yaml]
- archive 3.1.11 [crypto path]
- args 2.3.1
- async 2.8.2 [collection meta]
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.2.0
- charcode 1.3.1
- clock 1.1.0
- collection 1.16.0
- convert 3.0.1 [typed_data]
- coverage 1.2.0 [args logging package_config path source_maps stack_trace vm_service]
- crypto 3.0.1 [collection typed_data]
- fake_async 1.3.0 [clock collection]
- file 6.1.2 [meta path]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher material_color_utilities meta source_span stream_channel string_scanner term_glyph]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta vector_math]
- font_awesome_flutter 9.2.0 [flutter]
- frontend_server_client 2.1.3 [async path]
- fuchsia_remote_debug_protocol 0.0.0 [process vm_service file meta path platform]
- glob 2.0.2 [async collection file path string_scanner]
- google_sign_in_android 5.2.7 [flutter google_sign_in_platform_interface]
- google_sign_in_ios 5.2.7 [flutter google_sign_in_platform_interface]
- google_sign_in_platform_interface 2.1.3 [flutter quiver]
- google_sign_in_web 0.10.1+2 [flutter flutter_web_plugins google_sign_in_platform_interface js]
- http_multi_server 3.2.0 [async]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- intl 0.17.0 [clock path]
- io 1.0.3 [meta path string_scanner]
- js 0.6.4
- logging 1.0.2
- matcher 0.12.11 [stack_trace]
- material_color_utilities 0.1.4
- meta 1.7.0
- mime 1.0.2
- node_preamble 2.0.1
- package_config 2.0.2 [path]
- path 1.8.1
- platform 3.1.0
- plugin_platform_interface 2.1.2 [meta]
- pool 1.5.0 [async stack_trace]
- process 4.2.4 [file path platform]
- pub_semver 2.1.1 [collection meta]
- quiver 3.1.0 [matcher]
- shelf 1.3.0 [async collection http_parser path stack_trace stream_channel]
- shelf_packages_handler 3.0.0 [path shelf shelf_static]
- shelf_static 1.1.0 [convert http_parser mime path shelf]
- shelf_web_socket 1.0.1 [shelf stream_channel web_socket_channel]
- sky_engine 0.0.99
- source_map_stack_trace 2.1.0 [path stack_trace source_maps]
- source_maps 0.10.10 [source_span]
- source_span 1.8.2 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- sync_http 0.3.0
- term_glyph 1.2.0
- test_api 0.4.9 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph matcher]
- test_core 0.4.13 [analyzer async args boolean_selector collection coverage frontend_server_client glob io meta package_config path pool source_map_stack_trace source_maps source_span stack_trace stream_channel vm_service yaml matcher test_api]
- typed_data 1.3.0 [collection]
- url_launcher 6.1.2 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]
- url_launcher_android 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.0.16 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.0.5 [flutter plugin_platform_interface]
- url_launcher_web 2.0.11 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 3.0.1 [flutter url_launcher_platform_interface]
- vector_math 2.1.2
- vm_service 8.2.2
- watcher 1.0.1 [async path]
- web_socket_channel 2.2.0 [async crypto stream_channel]
- webdriver 3.0.0 [archive matcher path stack_trace sync_http]
- webkit_inspection_protocol 1.0.1 [logging]
- webview_flutter 3.0.4 [flutter webview_flutter_android webview_flutter_platform_interface webview_flutter_wkwebview]
- webview_flutter_android 2.8.8 [flutter webview_flutter_platform_interface]
- webview_flutter_platform_interface 1.8.1 [flutter plugin_platform_interface]
- webview_flutter_wkwebview 2.7.5 [flutter path webview_flutter_platform_interface]
- yaml 3.1.1 [collection source_span string_scanner]
```

</details>

---

**Comment by nilsreichardt** [comment]
@Lyokone Is this line causing the issue?

https://github.com/firebase/flutterfire/blob/da504308ec0979a9702cab894401887213ceebc0/packages/firebase_auth/firebase_auth/ios/Classes/FLTFirebaseAuthPlugin.m#L609

**Comment by darshankawar** [comment]
Thanks for the report. Using plugin example, seeing same behavior.

**Comment by Lyokone** [comment]
@nilsreichardt Thanks for the report, should be fixed with #9477

**Comment by nilsreichardt** [comment]
> @nilsreichardt Thanks for the report, should be fixed with #9477

Great, thanks for the quick fix 🙏
